CMakeLists.txt

```diff
@@ -41,39 +47,39 @@ ADD_EXECUTABLE(dump_json_ast dump_json_ast.cpp)
 TARGET_LINK_LIBRARIES(dump_json_ast graphqlparser)
 
 ADD_CUSTOM_COMMAND(
-  OUTPUT Ast.h
-  COMMAND python ./ast/ast.py c++ ./ast/ast.ast > Ast.h
-  DEPENDS ./ast/ast.ast
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Ast.h
+  COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/ast/ast.py c++ ${CMAKE_CURRENT_SOURCE_DIR}/ast/ast.ast > ${CMAKE_CURRENT_BINARY_DIR}/Ast.h
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/ast/ast.ast
```

Comment:
probably depend on ${CMAKE_CURRENT_SOURCE_DIR}/ast/ast.py and (repeat) on the next 4 blocks (also using a function here wouldn't be the worst idea)